AudiusProject/audius-protocol
# Use mediorum for storage v2 track upload

### Description

- Adds mediorum containers to protocol repo, gated by --storagev2 flag in `audius-compose up`
- Adds local /mediorum endpoint to match what's on staging and prod
- Adds v2 track upload skeleton to libs (will be moved to SDK later), pointed to the /mediorum endpoint

Caveat: Mediorum currently only supports dev (hardcoded to localhost) or stage/prod, so I need to sync with @stereosteve to see if we should allow the host to be configurable for dev or if it makes sense to set the env var to stage locally since it's pretty similar.


### Tests
- Make sure the AudiusProject/mediorum repo is cloned at the same level as audius-protocol and audius-client
- Start local stack: `audius-compose up -e 1 --no-libs --no-solana --storagev2` in audius-protocol. `npm run sdk:link` and `npm run web:dev` in audius-client
- Sign up on the client and set `storage_v2` feature flag to enabled (press "f" and scroll the the bottom)
- Upload a track and verify that it successfully sent a POST request to the /mediorum endpoint. You should be able to play the track at http://audius-protocol-storagev2-1/


### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
Only on local and gated behind the `storage_v2` feature flag.
